diff --git a/node_modules/@tanstack/solid-virtual/dist/cjs/index.cjs b/node_modules/@tanstack/solid-virtual/dist/cjs/index.cjs
index 7e97823..d75183f 100644
--- a/node_modules/@tanstack/solid-virtual/dist/cjs/index.cjs
+++ b/node_modules/@tanstack/solid-virtual/dist/cjs/index.cjs
@@ -39,7 +39,9 @@ function createVirtualizerBase(options) {
         (_a = options.onChange) == null ? void 0 : _a.call(options, instance2, sync);
       }
     }));
-    virtualizer.measure();
+    virtualizer._willUpdate();
+    setVirtualItems(store.reconcile(instance.getVirtualItems(), { key: "index" }));
+    setTotalSize(instance.getTotalSize());
   });
   return virtualizer;
 }
diff --git a/node_modules/@tanstack/solid-virtual/dist/esm/index.js b/node_modules/@tanstack/solid-virtual/dist/esm/index.js
index 1d52546..14c680a 100644
--- a/node_modules/@tanstack/solid-virtual/dist/esm/index.js
+++ b/node_modules/@tanstack/solid-virtual/dist/esm/index.js
@@ -38,7 +38,9 @@ function createVirtualizerBase(options) {
         (_a = options.onChange) == null ? void 0 : _a.call(options, instance2, sync);
       }
     }));
-    virtualizer.measure();
+    virtualizer._willUpdate();
+    setVirtualItems(reconcile(instance.getVirtualItems(), { key: "index" }));
+    setTotalSize(instance.getTotalSize());
   });
   return virtualizer;
 }
diff --git a/node_modules/@tanstack/solid-virtual/src/index.tsx b/node_modules/@tanstack/solid-virtual/src/index.tsx
index 69ac34f..9f16672 100644
--- a/node_modules/@tanstack/solid-virtual/src/index.tsx
+++ b/node_modules/@tanstack/solid-virtual/src/index.tsx
@@ -81,7 +81,9 @@ function createVirtualizerBase<
         },
       }),
     )
-    virtualizer.measure()
+    virtualizer._willUpdate()
+    setVirtualItems(reconcile(instance.getVirtualItems(), { key: 'index' }))
+    setTotalSize(instance.getTotalSize())
   })
 
   return virtualizer
