diff --git a/node_modules/solid-js/dist/dev.cjs b/node_modules/solid-js/dist/dev.cjs
index 7104749..0501eb1 100644
--- a/node_modules/solid-js/dist/dev.cjs
+++ b/node_modules/solid-js/dist/dev.cjs
@@ -764,6 +764,8 @@ function runComputation(node, value, time) {
     if (node.updatedAt != null && "observers" in node) {
       writeSignal(node, nextValue, true);
     } else if (Transition && Transition.running && node.pure) {
+      // On first computation during transition, also set committed value #2046
+      if (!Transition.sources.has(node)) node.value = nextValue;
       Transition.sources.add(node);
       node.tValue = nextValue;
     } else node.value = nextValue;
diff --git a/node_modules/solid-js/dist/dev.js b/node_modules/solid-js/dist/dev.js
index ea5e4bc..4b3ec07 100644
--- a/node_modules/solid-js/dist/dev.js
+++ b/node_modules/solid-js/dist/dev.js
@@ -762,6 +762,8 @@ function runComputation(node, value, time) {
     if (node.updatedAt != null && "observers" in node) {
       writeSignal(node, nextValue, true);
     } else if (Transition && Transition.running && node.pure) {
+      // On first computation during transition, also set committed value #2046
+      if (!Transition.sources.has(node)) node.value = nextValue;
       Transition.sources.add(node);
       node.tValue = nextValue;
     } else node.value = nextValue;
diff --git a/node_modules/solid-js/dist/solid.cjs b/node_modules/solid-js/dist/solid.cjs
index 7c133a2..2f68c99 100644
--- a/node_modules/solid-js/dist/solid.cjs
+++ b/node_modules/solid-js/dist/solid.cjs
@@ -717,6 +717,8 @@ function runComputation(node, value, time) {
     if (node.updatedAt != null && "observers" in node) {
       writeSignal(node, nextValue, true);
     } else if (Transition && Transition.running && node.pure) {
+      // On first computation during transition, also set committed value #2046
+      if (!Transition.sources.has(node)) node.value = nextValue;
       Transition.sources.add(node);
       node.tValue = nextValue;
     } else node.value = nextValue;
diff --git a/node_modules/solid-js/dist/solid.js b/node_modules/solid-js/dist/solid.js
index 656fd26..f08e9f5 100644
--- a/node_modules/solid-js/dist/solid.js
+++ b/node_modules/solid-js/dist/solid.js
@@ -715,6 +715,8 @@ function runComputation(node, value, time) {
     if (node.updatedAt != null && "observers" in node) {
       writeSignal(node, nextValue, true);
     } else if (Transition && Transition.running && node.pure) {
+      // On first computation during transition, also set committed value #2046
+      if (!Transition.sources.has(node)) node.value = nextValue;
       Transition.sources.add(node);
       node.tValue = nextValue;
     } else node.value = nextValue;
